### PR TITLE
JSON/gCP/DeepDiff v1.2 release tweaks

### PR DESCRIPTION
**Pull request by dgasmith**
## Description
A few additional tweaks for the Psi4 1.2 release.

- Updates gCP alias function testing so that basis sets are applied for aliases.
- Hides DeepDiff import so that *most* of Psi4 is usable without this module.
- Updates the JSON spec to the V1 release.
- Ensures QCSchema input is correctly *copied* and not just updated.

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: psi4/driver/json_wrapper.py [hunk 1]
```diff
@@ -82,9 +81,11 @@ def run_json(json_data, clean=True):
 
     # Attempt to run the computer
     try:
-        if ("schema_name" in json_data) and (json_data["schema_name"] == "QC_JSON"):
-            run_json_qc_schema(json_data, clean)
+        if ("schema_name" in json_data) and ("qc_schema" in json_data["schema_name"]):
```
Comment (loriab) [inline psi4/driver/json_wrapper.py]: nail it down further with `json_data["schema_name"].startswith("qc_schema")`?

File: psi4/driver/json_wrapper.py [hunk 2]
```diff
@@ -82,9 +81,11 @@ def run_json(json_data, clean=True):
...
+            # qc_schema should be copied
+            json_data = run_json_qc_schema(copy.deepcopy(json_data), clean)
```
Comment (loriab) [inline psi4/driver/json_wrapper.py]: +1

File: psi4/driver/json_wrapper.py [hunk 3]
```diff
@@ -123,7 +124,10 @@ def run_json_qc_schema(json_data, clean):
     _clean_psi_environ(clean)
 
     # This is currently a forced override
-    if json_data["schema_version"] != 0:
+    if json_data["schema_name"] != "qc_schema_input":
+        raise KeyError("Schema name of '{}' not understood".format(json_data["schema_name"]))
+
+    if json_data["schema_version"] != 1:
```
Comment (loriab) [inline psi4/driver/json_wrapper.py]: plan is to add to if/else here (and any other place schema processing forks) when want to run e.g., `1.dev5` through psi?

File: psi4/driver/procrouting/proc.py [hunk 4]
```diff
@@ -1216,9 +1216,9 @@ def scf_helper(name, post_scf=True, **kwargs):
             core.set_local_option('SCF', 'GUESS', 'CORE')
 
     if core.get_global_option('BASIS') == '':
-        if name == 'hf3c':
+        if name in ['hf3c', 'hf-3c']:
             core.set_global_option('BASIS', 'minix')
-        elif name == 'pbeh3c':
+        elif name in ['pbeh3c', 'pbeh-3c']:
```
Comment (loriab) [inline psi4/driver/procrouting/proc.py]: thanks, much saner.

File: tests/json/schema_1_orient/input.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import numpy as np
```
Comment (loriab) [inline tests/json/schema_1_orient/input.py]: regarding test names, any reason `schema_1_orient` can't go to `schema-1-orient`? I've gotten the whole test suite standardized on dashes, no underscores.

File: psi4/driver/json_wrapper.py (hunk 3 above)
Comment (loriab) [reply psi4/driver/json_wrapper.py]: sorry, `2.dev5`

File: tests/json/schema_1_orient/input.py (hunk 5 above)
Comment (loriab) [reply tests/json/schema_1_orient/input.py]: yes, please.

**Comment by dgasmith** [comment]
Roger that, all fixed I believe.

File: tests/json/schema-1-gradient/input.py [hunk 6]
```diff
@@ -57,7 +57,7 @@
   "return_energy": -76.02139738600329
 }
 
-psi4.json_wrapper.run_json(json_data)
+json_data = json_data = psi4.json_wrapper.run_json(json_data)
```
Comment (loriab) [inline tests/json/schema-1-gradient/input.py]: two `=`, huh?

File: tests/json/schema-1-properties/input.py [hunk 7]
```diff
@@ -107,12 +107,13 @@
 }
 
 
-psi4.json_wrapper.run_json(json_data)
+json_data = json_data = psi4.json_wrapper.run_json(json_data)
```
Comment (loriab) [inline tests/json/schema-1-properties/input.py]: ditto

File: tests/json/schema-1-gradient/input.py (hunk 6 above)
Comment (dgasmith) [reply tests/json/schema-1-gradient/input.py]: Something is going wonky somewhere. I dont see this on my laptop. Lets pull it in and check it out. It doesn't actually hurt anything, but we should patch it if we see it post-merge.

Comment (jturney) [reply tests/json/schema-1-gradient/input.py]: I see `json_data = json_data = psi4.json-wrapper....`

File: psi4/driver/json_wrapper.py [hunk 8]
```diff
@@ -133,6 +133,9 @@ def run_json_qc_schema(json_data, clean):
     if json_data["schema_version"] != 1:
         raise KeyError("Schema version of '{}' not understood".format(json_data["schema_version"]))
 
+    if json_data.get("ncores", False) is not False:
+        psi4.set_num_threads(json_data["ncores"], quiet=True)
```
[the remark this answers is not in the document]
Comment (dgasmith) [reply psi4/driver/json_wrapper.py]: Yup, fixed.

**Pull request merged**